napari/layers/_layer_actions.py

```diff
@@ -88,9 +92,19 @@ def _merge_stack(ll: LayerList, rgb=False):
     ll.append(merged)
 
 
-def _toggle_visibility(ll: LayerList):
+def _show_hide_selected(ll: LayerList):
+    global SHOW_SELECTED
     for lay in ll.selection:
-        lay.visible = not lay.visible
+        lay.visible = SHOW_SELECTED  # first time True, so will show
+    SHOW_SELECTED = not SHOW_SELECTED
+
+
+def _hide_show_unselected(ll: LayerList):
+    global SHOW_UNSELECTED
+    for lay in ll:
+        if lay not in ll.selection:
+            lay.visible = SHOW_UNSELECTED  # first time False, so will hide
+    SHOW_UNSELECTED = not SHOW_UNSELECTED
```

Comment:
Create an action that will receive `ViewerModel` instance, and the viewer will have additional attributes to control this instead of using global. 

Or add proper attribute to LayerList itself. 

The current solution will be terrible when use multiple viewers in one session.

Reply:
You mean move those two two variables to become attributes in the __init__ of LayerList?